File: src/main/java/QuadraticEquation_294.java
import java.util.Scanner;

public class QuadraticEquation_294 {
    public static void main(String[] args) {
        var scanner = new Scanner(System.in);
        var a = scanner.nextFloat();
        var b = scanner.nextFloat();
        var c = scanner.nextFloat();

        System.out.println(solve(a, b, c));
    }

    private static String solve(float a, float b, float c) {
        if (a == 0 && b == 0) return "IMPOSSIBLE";
        if (a == 0) return formatter(c / -b);
        if (b == 0) {
            if (a > 0 && c < 0) return formatter(-1f * (float) Math.sqrt(a)) + "\n" + formatter((float) Math.sqrt(a));
            else if (a < 0 && c > 0)
                return formatter(-1f * (float) Math.sqrt(a)) + "\n" + formatter((float) Math.sqrt(a));
            else return "IMPOSSIBLE";
        }
        if (c == 0) return "0.000" + "\n" + formatter(-1f * b / a);
        var delta = delta(a, b, c);
        if (delta < 0) return "IMPOSSIBLE";
        else if (delta == 0) return formatter(-b / (2 * a));
        else if (delta > 0) {
            var x1 = (-b + Math.sqrt(delta)) / (2 * a);
            var x2 = (-b - Math.sqrt(delta)) / (2 * a);
            return formatter((float) Math.min(x1, x2)) + "\n" + formatter((float) Math.max(x1, x2));
        } else return "IMPOSSIBLE";
    }

    private static String formatter(float x) {
        return String.format("%.3f", x);
    }

    private static float delta(float a, float b, float c) {
        return b * b - (4 * a * c);
    }
}

File: src/main/kotlin/QuadraticEquation_294.kt
import java.util.Scanner
import kotlin.math.sqrt

fun main() {
    val scanner = Scanner(System.`in`)
    val a = scanner.nextFloat()
    val b = scanner.nextFloat()
    val c = scanner.nextFloat()

    val x1: Float
    val x2: Float
    val delta = delta(a, b, c)
    if (a != 0f && b != 0f) {
        if (delta == 0f) {
            x1 = -1f * b / a * .5f
            printForQuera(x1)
        } else if (delta > 0) {
            x1 = ((-1f * b) + sqrt(delta)) / a * .5f
            x2 = ((-1f * b) - sqrt(delta)) / a * .5f
            printAscending(x1, x2)
        } else println("IMPOSSIBLE")
    } else {
        if (b == 0f) {
            if (c == 0f) printForQuera(0f)
            else if (a < 0 && c > 0) printAscending(sqrt(c / a), -1f * sqrt(c / a))
            else if (a > 0 && c < 0) printAscending(sqrt(c / a), -1f * sqrt(c / a))
            else println("IMPOSSIBLE")
        } else printForQuera(-1f * c / b)
    }
}

fun delta(a: Float, b: Float, c: Float) = (b * b) + (-4 * a * c)

fun Float.format() = "%.3f".format(this)

fun printAscending(x1: Float, x2: Float) {
    if (x1.isNaN() || x2.isNaN()) {
        println("IMPOSSIBLE")
    } else if (x1 == x2) {
        printForQuera(x1)
    } else if (x1 < x2) {
        printForQuera(x1)
        printForQuera(x2)
    } else {
        printForQuera(x2)
        printForQuera(x1)
    }
}

fun printForQuera(x: Float) {
    if (x == 0f) println("0.000")
    else println(x.format())
}